src/main/java/acme/entities/TrainingModule.java

```diff
@@ -0,0 +1,76 @@
+
+package acme.entities;
+
+import java.util.Date;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.ManyToOne;
+import javax.persistence.Temporal;
+import javax.persistence.TemporalType;
+import javax.validation.Valid;
+import javax.validation.constraints.AssertTrue;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Past;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Positive;
+
+import org.hibernate.validator.constraints.Length;
+import org.hibernate.validator.constraints.URL;
+
+import acme.client.data.AbstractEntity;
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+@Getter
+@Setter
+public class TrainingModule extends AbstractEntity {
+
+	private static final long	serialVersionUID	= 1L;
+
+	@NotBlank
+	@Column(unique = true)
+	@Pattern(regexp = "[A-Z]{1,3}-[0-9]{3}")
+	private String				code;
+
+	@Past
+	@NotNull
+	@Temporal(TemporalType.TIMESTAMP)
+	private Date				creationMoment;
+
+	@NotBlank
+	@Length(max = 100)
+	private String				details;
+
+	@NotNull
+	private DifficultyLevel		difficultyLevel;
+
+	@Past
+	@NotNull
+	@Temporal(TemporalType.TIMESTAMP)
+	private Date				updateMoment;
+
+	@URL
+	private String				link;
+
+	@NotNull
+	@Positive
+	private int					totalTime;
+
+
+	@AssertTrue(message = "Update moment must be after creation moment")
```

Comment:
This is so good, maybe it should be in the service, not here. Be sure of moving this to the service before the S03 deliverable.